distribution/Dockerfile

```diff
@@ -0,0 +1,35 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+FROM centos:7
+
+# Project version defined in pom.xml is passed as an argument
+ARG VERSION
+
+# JDK 8 is a pre-requisite to run Drill ; 'which' package is needed for drill-config.sh
+RUN yum install -y java-1.8.0-openjdk-devel which ; yum clean all ; rm -rf /var/cache/yum
+
+# The drill tarball is generated upon building the Drill project
+COPY target/apache-drill-$VERSION.tar.gz /tmp
+
+# Drill binaries are extracted into the '/opt/drill' directory
+RUN mkdir /opt/drill
+RUN tar -xvzf /tmp/apache-drill-$VERSION.tar.gz --directory=/opt/drill --strip-components 1
+
+# Starts Drill in embedded mode and connects to Sqlline
+ENTRYPOINT /opt/drill/bin/drill-embedded
```

Comment:
This `Dockerfile` is only for an embedded Drillbit, which is find for playing around.

Should we also offer an example production `Dockerfile`? In such a file, the following would also be needed:

* Copy in custom jars (UDFs custom storage plugins.)
* Copy in custom libraries (PAM, etc.)
* Specify custom config.

Doing the above will be easier if the user's files are not placed in the `$DRILL_HOME/conf` and other directories. That is, we want to use the site directory  feature.

Maybe allow a `/opt/drill-site` location and pass `--site /opt/drill-site` on the command line for the entry point.

Then, the `Dockerfile` can provide examples of how to copy the various kinds of files to the proper site directory locations:

* `/opt/drill-site/` - `drill-override.conf`, `core-site.xml`, `logback.xml`
* `/opt/drill-site/jar/` - UDF and storage-plugin jars
* `/opt/drill-site/lib/` - native libraries

Note that the functionality for Drill to find things in these site directory locations already exists. All we're doing here is showing the user how to use them.

Reply:
Hey Paul, thanks for the comments. This PR is to support Drill on Docker in embedded mode, to help people quickly get started with Drill. If they have Docker on their Mac or PC, they can just issue the Docker run command to pull the latest image of Drill and run in embedded mode. 

But, I agree that supporting a standalone Drill Dockerfile is necessary. I already have it working, but there is some more effort required - which i'll track in a different PR (tracked by a new JIRA). That would require documentation on how to pull in a ZK image (we already have it part of the Drill Docker Hub repo) and launch Drill in distributed mode. 

Regarding the drill site directory, that's a good suggestion - i'll try how to get that working seamlessly with Docker. Will reach out to you for your comments then.

Reply:
@paul-rogers, I meant to say that I have a separate ZK image which can be used to bring up ZK standalone, and use it for Drill. For example, see https://github.com/Agirish/drill-containers/blob/master/dockerfiles/apache-zookeeper-centos-3.4.10/Dockerfile

And yes, I agree having a consolidated image would be good. It will require us to bring up ZK (using the independent ZK image) and then bring up Drill using the ZK information. With K8S, it can be made seamless, but with Docker, it'll require some manual configurations. Hence pushed it out to a different PR. 

Regarding referencing it for the Drill book, may be we could connect over email. I have existing docs and instructions which i've used to successfully bring up Drill in distributed modes on Docker and K8S env. It could be faster than getting it into Apache :)